stores/navigationStore.ts

```diff
@@ -0,0 +1,35 @@
+import { create } from 'zustand';
+import type { Pages } from '../types';
+
+interface NavigationState {
+  prev_navigation_states: Pages[];
+  current_navigation_state: Pages;
+  updateNavigation: (page: Pages) => void;
+  revertToPreviousState: () => void;
+}
+
+export const useNavigationStore = create<NavigationState>((set) => ({
+  prev_navigation_states: [] as Pages[],
+  current_navigation_state: 'Login',
+  updateNavigation: (page) => {
+    set((state) => ({
+      ...state,
+      current_navigation_state: page,
+      prev_navigation_states:
+        state.prev_navigation_states.length >= 5
+          ? [page, ...state.prev_navigation_states.slice(0, 4)]
+          : [page, ...state.prev_navigation_states],
```

Comment:
I would just have an array and push to the end. If the history gets overwritten just set history to current state

Reply:
I am guessing there isn't an issue with this either. I think this is how zustand is intended to be used based off their documentation and some online examples